src/react/useBaseQuery.ts

```diff
@@ -96,7 +117,7 @@ export function useBaseQuery<TQueryFnData, TError, TData, TQueryData>(
     throw obsRef.current
       .fetchOptimistic(defaultedOptions)
       .then(({ data }) => {
-        defaultedOptions.onSuccess?.(data)
+        defaultedOptions.onSuccess?.(data as TData)
```

Comment:
@boschni I went a bit down the rabbithole with that last commit (395a5099), but I think types are now more exact than before. here, for example, `data` is inferred to `TData | undefined`, which is correct, and it was `any` before. But `onSuccess` expects `TData`, which is why I added the type assertion. I think this is the only real change - everything else is just drilling generics :)

Reply:
can we merge it then? I'd like to fix types of `initialData` but I don't want to start until this huge type monster is merged :)